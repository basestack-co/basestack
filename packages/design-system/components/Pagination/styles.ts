import styled from "styled-components";
import { space } from "styled-system";
import Text from "../Text";

export const Container = styled.div`
  ${space};
  display: flex;
  align-items: center;
`;

export const StyledText = styled(Text)`
  display: flex;
  font-weight: 500;
`;

export const Number = styled.div<{ highlight?: boolean }>`
  background-color: ${({ theme }) => theme.colors.gray100};
  height: 36px;
  min-width: 36px;
  padding: 0 10px;
  border-radius: 4px;
  display: flex;
  align-items: center;
  justify-content: center;
`;
