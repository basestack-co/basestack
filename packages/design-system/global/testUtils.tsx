import { type RenderOptions, render } from "@testing-library/react";
import type React from "react";
// Styles
import { ThemeProvider } from "styled-components";
import theme from "../theme/lightTheme";

interface Props {
  children: React.ReactNode;
}

const WithThemeProvider = ({ children }: Props) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

const renderWithTheme = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "wrapper">,
) => render(ui, { wrapper: WithThemeProvider, ...options });

export { renderWithTheme };
