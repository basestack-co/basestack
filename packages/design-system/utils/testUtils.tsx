import defaultTheme from "@basestack/design-system/theme/lightTheme";
import isPropValid from "@emotion/is-prop-valid";
import { RenderResult, render } from "@testing-library/react";
import React from "react";
import { StyleSheetManager, ThemeProvider } from "styled-components";

const Provider = ({
  children,
  theme = defaultTheme,
}: {
  children: React.ReactNode;
  theme: any;
}) => (
  <ThemeProvider theme={theme}>
    <StyleSheetManager shouldForwardProp={isPropValid}>
      {children}
    </StyleSheetManager>
  </ThemeProvider>
);

export const renderWithTheme = (
  ui: React.ReactElement,
  options?: any,
): RenderResult => render(ui, { wrapper: Provider, ...options });
