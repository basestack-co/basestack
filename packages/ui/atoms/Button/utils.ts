import { Size, Target, Type, Variant } from "@/components/atoms/Button/types";
import { css, DefaultTheme } from "styled-components";
import { darken, rem, transparentize } from "polished";

export const handleButtonSize = (size?: Size) => {
  switch (size) {
    case "large":
      return css`
        height: ${rem("60px")};
        font-size: ${rem("18px")};
        line-height: ${rem("25px")};
        padding: 0 ${rem("40px")};
      `;
    case "medium":
      return css`
        height: ${rem("54px")};
        font-size: ${rem("16px")};
        line-height: ${rem("22px")};
        padding: 0 ${rem("34px")};
      `;
    case "normal":
      return css`
        height: ${rem("48px")};
        font-size: ${rem("16px")};
        line-height: ${rem("22px")};
        padding: 0 ${rem("30px")};
      `;
    default:
    case "small":
      return css`
        height: ${rem("46px")};
        font-size: ${rem("14px")};
        line-height: ${rem("19px")};
        padding: 0 ${rem("24px")};
      `;
    case "xSmall":
      return css`
        height: ${rem("42px")};
        font-size: ${rem("14px")};
        line-height: ${rem("19px")};
        padding: 0 ${rem("20px")};
      `;
  }
};

export const handleButtonsColors = (theme: DefaultTheme) => ({
  primary: {
    outline: {
      color: theme.colors.primary,
      borderColor: theme.colors.primary,
      backgroundColor: "transparent",
    },
    filled: {
      color: theme.colors.white,
      borderColor: "transparent",
      backgroundColor: theme.colors.primary,
    },
  },
  secondary: {
    outline: {
      color: theme.colors.secondary,
      borderColor: theme.colors.secondary,
      backgroundColor: "transparent",
    },
    filled: {
      color: theme.colors.white,
      borderColor: "transparent",
      backgroundColor: theme.colors.secondary,
    },
  },
  tertiary: {
    outline: {
      color: theme.colors.gray3,
      borderColor: theme.colors.gray3,
      backgroundColor: "transparent",
    },
    filled: {
      color: theme.colors.primary,
      borderColor: "transparent",
      backgroundColor: theme.colors.gray6,
    },
  },
  neutral: {
    outline: {
      color: theme.colors.gray2,
      borderColor: theme.colors.gray2,
      backgroundColor: "transparent",
    },
    filled: {
      color: theme.colors.primary,
      borderColor: "transparent",
      backgroundColor: "transparent",
    },
  },
  success: {
    outline: {
      color: theme.colors.success,
      borderColor: theme.colors.success,
      backgroundColor: "transparent",
    },
    filled: {
      color: theme.colors.white,
      borderColor: "transparent",
      backgroundColor: theme.colors.success,
    },
  },
  warning: {
    outline: {
      color: theme.colors.warning,
      borderColor: theme.colors.warning,
      backgroundColor: "transparent",
    },
    filled: {
      color: theme.colors.white,
      borderColor: "transparent",
      backgroundColor: theme.colors.warning,
    },
  },
  danger: {
    outline: {
      color: theme.colors.error,
      borderColor: theme.colors.error,
      backgroundColor: "transparent",
    },
    filled: {
      color: theme.colors.white,
      borderColor: "transparent",
      backgroundColor: theme.colors.error,
    },
  },
});

export const darkenColors = (
  theme: DefaultTheme,
  amount: number,
  type: Type,
  variant: Variant,
  target: Target
) => {
  return darken(
    amount || 0.1,
    handleButtonsColors(theme)[type || "primary"][variant || "filled"][target]
  );
};

export const lightenColors = (
  theme: DefaultTheme,
  amount: number,
  type: Type,
  variant: Variant,
  target: Target
) => {
  return transparentize(
    amount || 0.6,
    handleButtonsColors(theme)[type || "primary"][variant || "filled"][target]
  );
};
