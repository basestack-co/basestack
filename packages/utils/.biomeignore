# Node dependencies
node_modules

# Next.js build output
.next
out

# Environment files
.env
.env.*
!.env.example

# Logs
*.log
npm-debug.log*
yarn-debug.log*
yarn-error.log*
pnpm-debug.log*

# OS / editor files
.DS_Store
Thumbs.db

# Package manager cache
.pnp
.pnp.js
.pnp.cjs

# Test coverage
coverage

# Build artifacts
dist
build

# Temporary files
.tmp
temp