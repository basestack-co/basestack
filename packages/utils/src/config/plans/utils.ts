export const getAppMode = (mode: string) => {
  return (
    ["development", "production", "local"].includes(mode) ? mode : "production"
  ) as "local" | "development" | "production";
};
