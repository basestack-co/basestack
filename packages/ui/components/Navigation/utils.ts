// Types
import { Translate } from "next-translate";
import { NextRouter } from "next/router";
// Utils
import { config } from "@basestack/utils";
import { signOut } from "next-auth/react";

export const getInternalLinks = (t: Translate, formId: string) => {
  return [
    {
      icon: "flag",
      activeText: "Submissions",
      to: `/form/${formId}/submissions`,
      text: t("internal.submissions"),
    },
    {
      icon: "flag",
      activeText: "Setup",
      to: `/form/${formId}/setup`,
      text: t("internal.setup"),
    },
    {
      icon: "settings",
      activeText: "Settings",
      to: `/form/${formId}/settings/general`,
      text: t("internal.settings"),
    },
  ];
};

export const getExternalLinks = (t: Translate) => {
  return [
    {
      icon: "description",
      activeText: "Documentation",
      to: config.urls.docs.flags.base,
      isExternal: true,
      text: t("external.docs"),
    },
    {
      icon: "link",
      activeText: "Github",
      to: config.urls.repo,
      isExternal: true,
      text: t("external.github"),
    },
  ];
};

export const getAvatarDropdownList = (
  t: Translate,
  router: NextRouter,
  onCreateForm: () => void,
) => {
  return [
    {
      icon: "add_circle",
      text: t("create.form"),
      onClick: onCreateForm,
      separator: true,
    },
    {
      icon: "settings",
      text: t("dropdown.settings"),
      onClick: () =>
        router.push({
          pathname: "/user/profile/settings",
        }),
    },
    {
      icon: "logout",
      text: t("dropdown.logout"),
      onClick: signOut,
    },
  ];
};
