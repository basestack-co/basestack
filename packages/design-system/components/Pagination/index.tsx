import { memo } from "react";
import { SpaceProps } from "styled-system";
import { useTheme } from "styled-components";
import { Button, ButtonVariant } from "../Button";
import { Container, Number, StyledText } from "./styles";

export interface PaginationProps extends SpaceProps {
  /**
   * onClick load more
   */
  onClick: () => void;
  /**
   * current page
   */
  currentPage: number;
  /**
   * number of pages
   */
  totalPages: number;
  /**
   * show loading spinner
   */
  isLoading: boolean;
}

const Pagination = ({
  onClick,
  currentPage = 1,
  totalPages = 1,
  isLoading = false,
  ...props
}: PaginationProps) => {
  const theme = useTheme();

  return (
    <Container data-testid="pagination" {...props}>
      <StyledText
        data-testid="pagination-current-page"
        size="small"
        muted
        mr={theme.spacing.s1}
      >
        Showing
      </StyledText>
      <Number>
        <StyledText data-testid="pagination-current-page" size="small">
          {currentPage}
        </StyledText>
      </Number>
      <StyledText
        data-testid="pagination-total-pages"
        ml={theme.spacing.s1}
        size="small"
        muted
      >
        of {""}
        {totalPages}
      </StyledText>
      <Button
        onClick={onClick}
        variant={ButtonVariant.Neutral}
        ml={theme.spacing.s1}
        isLoading={isLoading}
        isDisabled={currentPage >= totalPages}
      >
        Load More
      </Button>
    </Container>
  );
};

export default memo(Pagination);
