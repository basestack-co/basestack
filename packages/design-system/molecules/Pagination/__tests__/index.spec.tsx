import React from "react";
import { cleanup } from "@testing-library/react";
import { renderWithTheme } from "../../../utils/testUtils";
import Pagination from "..";

describe("Pagination Molecule tests", () => {
  afterEach(cleanup);

  test("should render Pagination correctly", () => {
    const { asFragment } = renderWithTheme(
      <Pagination
        currentPage={10}
        totalPages={11}
        onNext={jest.fn()}
        onPrev={jest.fn()}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("should render Pagination correctly", () => {
    const { getByTestId, getByText } = renderWithTheme(
      <Pagination
        currentPage={10}
        totalPages={11}
        onNext={jest.fn()}
        onPrev={jest.fn()}
      />
    );
    const prevButton = getByText(/Prev/);
    const nextButton = getByText(/Next/);
    const currentPage = getByTestId("pagination-current-page");
    const totalPages = getByTestId("pagination-total-pages");

    expect(prevButton).toBeVisible();
    expect(nextButton).toBeVisible();
    expect(currentPage).toHaveTextContent("10");
    expect(totalPages).toHaveTextContent("of 11");
  });
});
