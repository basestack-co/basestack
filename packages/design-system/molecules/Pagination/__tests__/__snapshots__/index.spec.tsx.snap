// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Molecule tests should render Pagination correctly 1`] = `
<DocumentFragment>
  .c5 {
  font-size: 0.875rem;
  line-height: 1.375rem;
  font-weight: 400;
  font-family: 'Roboto',sans-serif;
  color: #000000;
}

.c7 {
  font-size: 0.875rem;
  line-height: 1.375rem;
  font-weight: 400;
  margin-left: 0.75rem;
  font-family: 'Roboto',sans-serif;
  color: #000000;
}

.c2 {
  height: 1.5rem;
  width: 1.5rem;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin-right: 0.5rem;
}

.c9 {
  height: 1.5rem;
  width: 1.5rem;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin-left: 0.5rem;
}

.c3 {
  color: #000000;
  font-size: 1.5rem;
}

.c1 {
  border: none;
  height: 2.25rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 0.75rem;
  font-size: 0.875rem;
  font-weight: 500;
  cursor: pointer;
  border-radius: 4px;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: background-color 0.1s ease-in-out,color 0.1s ease-in-out, border-color 0.1s ease-in-out;
  transition: background-color 0.1s ease-in-out,color 0.1s ease-in-out, border-color 0.1s ease-in-out;
  background-color: transparent;
  color: #000000;
  margin-right: 0.75rem;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  padding-left: 0.5rem;
}

.c1:hover:not(:disabled) {
  background-color: #EEEEEE;
}

.c1:active:not(:disabled) {
  background-color: #E2E2E2;
}

.c1 .material-symbols-sharp {
  color: inherit;
}

.c1:disabled {
  cursor: not-allowed;
}

.c8 {
  border: none;
  height: 2.25rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 0.75rem;
  font-size: 0.875rem;
  font-weight: 500;
  cursor: pointer;
  border-radius: 4px;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: background-color 0.1s ease-in-out,color 0.1s ease-in-out, border-color 0.1s ease-in-out;
  transition: background-color 0.1s ease-in-out,color 0.1s ease-in-out, border-color 0.1s ease-in-out;
  background-color: transparent;
  color: #000000;
  margin-left: 0.75rem;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  padding-right: 0.5rem;
}

.c8:hover:not(:disabled) {
  background-color: #EEEEEE;
}

.c8:active:not(:disabled) {
  background-color: #E2E2E2;
}

.c8 .material-symbols-sharp {
  color: inherit;
}

.c8:disabled {
  cursor: not-allowed;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-weight: 500;
}

.c4 {
  background-color: #EEEEEE;
  height: 36px;
  min-width: 36px;
  padding: 0 10px;
  border-radius: 4px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div
    class="c0"
    data-testid="pagination"
  >
    <button
      class="c1"
    >
      <div
        class="c2"
        data-testid="icon-container"
        size="24"
      >
        <span
          class="c3 material-symbols-sharp"
          data-testid="icon"
          size="24"
        >
          chevron_left
        </span>
      </div>
      Prev
    </button>
    <div
      class="c4"
    >
      <p
        class="c5 c6"
        data-testid="pagination-current-page"
      >
        10
      </p>
    </div>
    <p
      class="c7 c6"
      data-testid="pagination-total-pages"
    >
      of 11
    </p>
    <button
      class="c8"
    >
      Next
      <div
        class="c9"
        data-testid="icon-container"
        size="24"
      >
        <span
          class="c3 material-symbols-sharp"
          data-testid="icon"
          size="24"
        >
          chevron_right
        </span>
      </div>
    </button>
  </div>
</DocumentFragment>
`;
